dht_test.go

```diff
@@ -1401,12 +1407,12 @@ func testFindPeerQuery(t *testing.T,
 	for i := 0; i < bootstrappers; i++ {
 		for j := 0; j < bootstrapperLeafConns; j++ {
 			v := mrand.Intn(leafs)
-			connect(t, ctx, others[i], others[bootstrappers+v])
+			connectNoSync(t, ctx, others[i], others[bootstrappers+v])
 		}
 	}
 
 	for i := 0; i < bootstrappers; i++ {
-		connect(t, ctx, guy, others[i])
+		connectNoSync(t, ctx, guy, others[i])
```

Comment:
@aschmahmann The thing is, the assumption that `connected==RT.member` dosen't hold anymore and so this could block indefinitely if `fixLowPeers` dosen't trigger on a connection because there are enough peers in the RT. I've added a time.Sleep after these calls for things to settle down/RT to have some members and then we call `RefreshRoutingTable` on all of them. Things work well now:

```
aarshshah@Aarshs-MacBook-Pro go-libp2p-kad-dht % GO111MODULE=on go test -count=50 -race -run ^TestFindPeerQuery
PASS
ok  	github.com/libp2p/go-libp2p-kad-dht	162.057s
```

Reply:
The thing is, because `connected != rt.Member`, the wait could hang indefinitely here if there are more peers we want to want on than `minRTPeerThreshold`.

 I've added a short sleep after these calls for things to settle down/RT to have some members before we kick off the refresh.

Reply:
> I've added a short sleep after these calls for things to settle down/RT

ugh. We do have these sleeps all over the code base, so I'm not going to say no here, but can we not use a bunch of `wait(t,ctx, dht1,dht2)` or `waitForWellFormedTables` here instead? The peers should always have at least a few of the bootstrappers in their routing tables (definitely at least the `minRTRefreshThreshold = 10`, but probably as high as a full bucket as well).

Can do these in a followup issue + PR if you'd prefer.

Reply:
Have made the change.